dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/report/TrackerErrorReport.java

```diff
@@ -104,26 +108,60 @@ public TrackerErrorReportBuilder addArg( Object arg )
             return this;
         }
 
+        public TrackerErrorReportBuilder addArg( String arg )
+        {
+            this.arguments.add( arg );
+            return this;
+        }
+
+        public TrackerErrorReportBuilder addArg( Instant instant )
+        {
+            // TODO EnrollmendDateValidationHook uses E1025 and E1025 for
+            // malformed and null occuredAt, enrolledAt
+            // EventDateValidationHook uses errors like E1031 without args to
+            // report a required instant that is null.
+            if ( instant == null )
+            {
+                this.arguments.add( "" );
+                return this;
+            }
+            this.arguments.add( DateFormat.getInstance().format( Date.from( instant ) ) );
+            return this;
+        }
+
+        public TrackerErrorReportBuilder addArg( Date date )
+        {
+            this.arguments.add( DateFormat.getInstance().format( date ) );
+            return this;
+        }
+
+        public TrackerErrorReportBuilder addArg( TrackerIdScheme idScheme, IdentifiableObject arg )
+        {
+            final TrackerIdentifier identifier = TrackerIdentifier.builder().idScheme( idScheme ).build();
+            this.arguments.add( identifier.getIdAndName( arg ) );
+            return this;
+        }
+
+        public TrackerErrorReportBuilder addArg( TrackerDto dto )
+        {
+            this.arguments.add( dto.getClass().getSimpleName() + " (" + dto.getUid() + ")" );
```

Comment:
It might not be relevant, but this assumes uid is present/used. The idea about IdScheme is that we can use other properties for identifying data, so data might be passed in without uid, but CODE or ATTRIBUTE+value as a replacement for uid. Would it be a big step to use the idScheme of the import here to report the identifier?

Reply:
I am going to set the comments to resolved as I think we should tackle this in its own issue.

Some thoughts on it: the existing `addArg(TrackerIdScheme, IdentifiableObject)` shows how this can be done. This would mean code creating errors need to pass in the TrackerIdScheme specified in the bundle. We could investigate what the cleanest way would be. Maybe the TrackerValidationReport could be made responsible for this. As on its creation we know for what TrackerBundle import the validation will be done. TrackerBundle also has the TrackerIdScheme information.